data/carepartner/alerts/alerts.go

```diff
@@ -0,0 +1,85 @@
+package alerts
+
+import (
+	"net/http"
+
+	"github.com/tidepool-org/platform/data/service"
+	"github.com/tidepool-org/platform/log"
+)
+
+// List the care partner alerts.
+func List(sCtx service.Context) {
+	ctx := sCtx.Request().Context()
+	lgr := log.LoggerFromContext(ctx)
+
+	lgr.Info("list the care partner alerts")
+	sCtx.RespondWithStatusAndData(http.StatusNotImplemented, nil)
+}
+
+// Get a specific care partner alert.
+func Get(sCtx service.Context) {
+	ctx := sCtx.Request().Context()
+	lgr := log.LoggerFromContext(ctx)
+
+	lgr.Info("get a specific care partner alert")
+	sCtx.RespondWithStatusAndData(http.StatusNotImplemented, nil)
+}
+
+// Create a new care partner alert.
+func Create(sCtx service.Context) {
+	ctx := sCtx.Request().Context()
+	lgr := log.LoggerFromContext(ctx)
+
+	lgr.Info("create a care partner alert")
+	sCtx.RespondWithStatusAndData(http.StatusNotImplemented, nil)
+}
+
+// Update a care partner alert.
+func Update(sCtx service.Context) {
+	ctx := sCtx.Request().Context()
+	lgr := log.LoggerFromContext(ctx)
+
+	lgr.Info("update a care partner alert")
+	sCtx.RespondWithStatusAndData(http.StatusNotImplemented, nil)
+}
+
+// Ack acknowledges an alert.
+func Ack(sCtx service.Context) {
+	ctx := sCtx.Request().Context()
+	lgr := log.LoggerFromContext(ctx)
+
+	lgr.Info("create an alert acknowledgement")
+	sCtx.RespondWithStatusAndData(http.StatusNotImplemented, nil)
+}
+
+// CreateAppleTokens stores a token used to push alerts to Apple devices.
+func CreateAppleToken(sCtx service.Context) {
+	ctx := sCtx.Request().Context()
+	lgr := log.LoggerFromContext(ctx)
+
+	lgr.Info("store an Apple push notification token")
+	sCtx.RespondWithStatusAndData(http.StatusNotImplemented, nil)
+}
+
+// AuthWrapper lets the base service inject authentication.
+//
+// Cuz alerts don't care how users are authenticated.
+type AuthWrapper = func(service.HandlerFunc) service.HandlerFunc
```

Comment:
The comment I left for §PathBuilder§ applies here too. I think it's better if we are explicit regarding who's authorized to access those endpoints, so it's obvious for the reader.

Reply:
I guess I'd have to re-arch some things then. It'd be my preference to keep the alerts routes in the alerts package, but in order to use the existing `Authenticate()`, I'd have to move the alerts routes into the v1 package. Which feels like a coupling I'd prefer to avoid. That's the "real" reason for the AuthWrapper. As this is all one codebase, I'm not really saying that alerts shouldn't have to care how the authentication is being done, but it also shouldn't be reinventing that, but at the same time, with the way the v1 package is architected, I can't use `Authenticate()` from alerts, without creating a circular dependency.

Would you be opposed to lifting the `Authenticate()` out of v1 if I could find a reasonable way? That would let me keep the routes for alerts in the alerts package, and still have an explicit auth, granted it would still be in a different package, but at least which package would be clearly called out (as the function call would be something like `v1auth.Authenticate()` or so.)

Reply:
Re-reading this myself, I don't care about the routes being in the alerts package... that's a separation I can live with. It's almost better... they're just handlers, they don't know about routes. I can grok that.

Reply:
You can use `service/api/RequireUser`/`service/api/RequireServer`/`service/api/Require`. I'm not sure why `data` has its own authentication middleware.

Reply:
It has a slightly different signature, but it's doing basically the same thing as `service/api/Require`. I think it should be calling `Require` under the hood, and could probably be modified to operate that way, but I'll handle that in a separate PR unless you don't think it's worth messing with?